core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -110,6 +110,19 @@ void SYCL::print_configuration(std::ostream& os, bool verbose) const {
 #else
   os << "macro  KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES : undefined\n";
 #endif
+#ifdef SYCL_EXT_INTEL_QUEUE_IMMEDIATE_COMMAND_LIST
+  if (sycl_queue()
+          .has_property<
+              sycl::ext::intel::property::queue::immediate_command_list>())
+    os << "Immediate command lists enforced\n";
+  else if (sycl_queue()
+               .has_property<sycl::ext::intel::property::queue::
+                                 no_immediate_command_list>())
+    os << "Standard command queue enforced\n";
+  else
+#endif
+    os << "Immediate command lists and standard command queue allowed. "
+          "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS takes precedence.\n";
```

Comment:
Is that an environment variable?  If so you might want to do `std::getenv()` and spell out if it is defined and wha the value is.